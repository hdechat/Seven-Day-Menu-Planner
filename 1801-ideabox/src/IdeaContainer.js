import React from 'react';
import { Card } from './Card.js';
import PropTypes from 'prop-types';

const IdeaContainer = ({ ideas, removeIdea }) => {

  const ideaCards = ideas.map((idea, index) => 
    <Card 
      key={index}
      {...idea} //gives all the propertys of idea. easier editing: if this.state in app changes, all changes follow.
      removeIdea={removeIdea}
    />);

  return(
    <div>
      {ideaCards}
    </div>
  )
}

IdeaContainer.propTypes = {
  ideas: PropTypes.array.isRequired,
  removeIdea: PropTypes.func.isRequired
}

export default IdeaContainer;