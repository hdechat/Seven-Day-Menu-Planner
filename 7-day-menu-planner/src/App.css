body {
  background-repeat: no-repeat;
  box-sizing: border-box;
}

h1 {
}

.App {
  text-align: center;
}

.App-header {
  background: #878984;
  display: flex;
  justify-content: space-around;
  height: 2rem;
  padding: 0rem 30rem 0.5rem 20rem;
}

.top-of-nav {
  margin: -2px;
}

.bottom-of-nav {
  margin: 6px;
}

.nav {
  align-self: baseline;
  color: #fff;
  font-size: 1.2rem;
  margin: 0;
  padding: 0.5rem 3rem;
  text-decoration: none;
}

.nav:hover {
  color: #9bd437;
}

.title {
  background: #878984;
  color: #9bd437;
  text-decoration: none;
  float: left;
  font-size: 1.5rem;
  margin: 0;
  padding-left: 5px;
  line-height: 39px;
}

.welcome-message {
  color: #de3400;
  font-size: 50px;
}

@media screen and (max-width: 350px) {
  body {
    background: #ADC3D0;
  }

  .App-header {
    height: 5rem;
    margin-top: -1.5rem;
    width: fit-content;
  }

  .title {
    display: none;
  }

  .nav {
    margin: 2rem 1.7rem 2rem 2rem;
    font-size: 1.5rem;
  }

  .english-spanish {
      margin: auto 17rem;
  }

  .select-category {
    height: 3rem;
    margin: 1rem;
    width: 14rem;
  }

  .search {
    font-size: 20px;
    width: 14rem;
    height: 3rem;
  }

  .search::placeholder {
    font-size: 20px;
  }

  .welcome-message {
    font-size: 40px;
    width: 36rem;
  }

  .display-cards {
    overflow-y: unset;
    width: 180%;
  }

  select {
    font-size: 20px;
    height: 3rem;
  }
}
