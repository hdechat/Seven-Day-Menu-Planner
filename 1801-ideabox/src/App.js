import React, { Component } from 'react';
import IdeaForm from './IdeaForm.js';
import IdeaContainer from './IdeaContainer.js';
import './App.css';

class App extends Component {
  constructor(props) { //allows access to lifecycle methods
    super(props);

    this.state = {
      ideas: []
    }
  }

  addIdea = (idea) => {
    const newIdea = {...idea, id: Date.now()}
    const ideas = [newIdea, ...this.state.ideas]
    this.setState({ ideas })
  }

  removeIdea = (id) => {
    const filteredIdeas = this.state.ideas.filter(idea => idea.id !== id);
    this.setState({ideas: filteredIdeas});
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1 className="App-title">IdeaBox</h1>
          <IdeaForm addIdea={this.addIdea}/>
        </header>
        <IdeaContainer ideas={this.state.ideas} removeIdea={this.removeIdea}/>
      </div>
    );
  }
}

export default App;
