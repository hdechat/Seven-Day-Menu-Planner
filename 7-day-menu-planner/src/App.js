import React, { Component } from 'react';
import './App.css';
import { NavLink, Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { chooseCategory } from './actions';
import RecipeCardsDisplay from './components/RecipeCardsDisplay';
import MenuCalendar from './components/MenuCalendar';

export class App extends Component {

  displayLoadingMessage() {
    return this.props.recipesIsLoading ?
      <p className="loading-message">Your Recipes Are Loading</p> : null;
  }

  render() {
    const { recipeCards, chooseCategory } = this.props;

    return (
      <div className="App">
        <header>
          <NavLink to='/menu-calendar'>Menu Calendar</NavLink>
          <Route exact path='/menu-calendar' component={MenuCalendar} />
        </header>
        <select 
          onChange={event => chooseCategory(event.target.value)}
          id='category'>
          <option>SELECT CATEGORY</option>
          <option value='paleo'>Paleo</option>
          <option value='vegetarian'>Vegetarian</option>
          <option value='vegan'>Vegan</option>
        </select>
        {this.displayLoadingMessage()}
        <RecipeCardsDisplay recipeCards={recipeCards} />
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  recipesIsLoading: state.recipes.recipesIsLoading,
  recipeCards: state.recipes.results
});
export const mapDispatchToProps = dispatch => ({
  chooseCategory: category => dispatch(chooseCategory(category))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
