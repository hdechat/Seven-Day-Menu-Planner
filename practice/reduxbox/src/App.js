import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import TodoForm from './components/TodoForm.js'
import TodoList from './components/TodoList.js'
import { connect } from 'react-redux'
import { addTodo } from './actions'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <main className="App-intro">
          <TodoForm this.props.handleSubmit/>
          <TodoList todoList={this.props.todoList}/>
        </main>
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  todoList: state.todoList
})

const mapDispatchToProps = (dispatch) => ({
  handleSubmit: (todo) => (dispatch(addTodo(todo)))
})

connect(mapStateToProps, mapDispatchToProps)(App)

export default App;
