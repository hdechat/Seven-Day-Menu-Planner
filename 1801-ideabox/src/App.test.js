import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import IdeaContainer from './IdeaContainer';
import renderer from 'react-test-renderer';

import { shallow, mount } from 'enzyme';

//component is rendered
//functions execute
//component is rendered with correct props
//component calls a function
//defaults

//1.Setup
//2.Execution
//3.Assertion
//4.optional in JS - Tear Down

describe('App', () => {
  it('renders', () => {
    const app = shallow(<App />);

    expect(app).toHaveLength(1);
  });

  it('renders an IdeaContainer with the correct props', () => {
    const app = shallow(<App />);

    expect(app.find(IdeaContainer).prop('ideas')).toEqual([]);
  });

  it('matches the snapshot', () => {
      //creates snapshot which can be found int node_module file.  append with toJSON() to console.log more clearly
    const app = renderer.create(<App />);

    expect(app).toMatchSnapshot();
  });

  describe('removeIdea', () => {
    it('changes state', () => {
      const app = shallow(<App />);
      const id = 3;

      //set state on wrapper to set up state to create idea with the id of '3' in order to set up test for method
      // app.setState([])

      app.instance().removeIdea(id);

      expect()


    });
  });
});
