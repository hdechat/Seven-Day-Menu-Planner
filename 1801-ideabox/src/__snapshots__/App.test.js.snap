// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      IdeaBox
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="title"
        onChange={[Function]}
        type="text"
        value=""
      />
      <input
        name="body"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </header>
  <div />
</div>
`;
